src/main/java/org/opensearch/knn/index/KNNVectorDVLeafFieldData.java

```diff
@@ -48,6 +50,44 @@ public ScriptDocValues<float[]> getScriptValues() {
 
     @Override
     public SortedBinaryDocValues getBytesValues() {
-        throw new UnsupportedOperationException("knn vector field '" + fieldName + "' doesn't support sorting");
+        try {
+            final BinaryDocValues binaryDocValues = DocValues.getBinary(reader, fieldName);
+            SortedBinaryDocValues sortedBinaryDocValues = new SortedBinaryDocValues() {
+
+                private boolean docExists = false;
+                float[] floats = null;
+                int pos = 0;
+                BytesRefBuilder bytesRefBuilder = new BytesRefBuilder();
+
+                @Override
+                public boolean advanceExact(int doc) throws IOException {
+                    if (binaryDocValues.advanceExact(doc)) {
+                        docExists = true;
+                        floats = vectorDataType.getVectorFromDocValues(binaryDocValues.binaryValue());
+                        pos = 0;
+                        return docExists;
+                    }
+                    docExists = false;
+                    return docExists;
+                }
+
+                @Override
+                public int docValueCount() {
+                    return docExists ? floats.length : 1;
+                }
+
+                @Override
+                public BytesRef nextValue() throws IOException {
+                    Float v = floats[pos++];
+                    bytesRefBuilder.clear();
+                    bytesRefBuilder.copyChars(v.toString());
+                    return bytesRefBuilder.get();
+                }
+            };
+            return sortedBinaryDocValues;
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
please use logger here to log the exception. But do we want to log the exception or throw the exception?

Reply:
FIXED at https://github.com/opensearch-project/k-NN/pull/1571/commits/f4a69ca2f04833f238a54d76e75a9ed4f3db0dcb